src/ralph/backends/database/mongo.py

```diff
@@ -0,0 +1,128 @@
+"""MongoDB database backend for Ralph"""
+
+import hashlib
+import json
+import logging
+from typing import Generator, TextIO, Union
+
+from bson.objectid import ObjectId
+from pymongo import MongoClient
+from pymongo.errors import BulkWriteError
+
+from ralph.defaults import MONGO_COLLECTION, MONGO_DATABASE, MONGO_URI
+
+from .base import BaseDatabase
+
+logger = logging.getLogger(__name__)
+
+
+class MongoDatabase(BaseDatabase):
+    """Mongo database backend."""
+
+    name = "mongo"
+
+    def __init__(
+        self,
+        connection_uri: str = MONGO_URI,
+        database: str = MONGO_DATABASE,
+        collection: str = MONGO_COLLECTION,
+        client_options: dict = None,
+    ):
+        """Instanciates the Mongo client.
+
+        Args:
+            connection_uri (str): MongoDB connection URI.
+            database (str): MongoDB database to connect to.
+            collection (str): MongoDB database collection to get objects from.
+            client_options (dict): A dictionary of valid options for the MongoClient
+                class initialization.
+        """
+        if client_options is None:
+            client_options = {}
+
+        self.client = MongoClient(connection_uri, **client_options)
+        self.database = getattr(self.client, database)
+        self.collection = getattr(self.database, collection)
+
+    def get(self, chunk_size: int = 500):
+        """Gets collection documents and yields them."""
+
+        for document in self.collection.find(batch_size=chunk_size):
+            # Make the document json-serializable
+            document.update({"_id": str(document.get("_id"))})
+            yield document
```

Comment:
> This is true when the xAPI statement is stored at the root of the document, however, LearningLocker, as you mentioned, stores it inside a "statement" field which has next to it a "person" field containing another MongoDB _id field.

I think you are making a point that needs to be discussed furthermore about the "schema" we use to store statements in the MongoDB backend. Let's do this IRL!

> Alternatively, we could select only the "statement" field in the find query to only work with xAPI data in the case of LearningLocker, however, this would make the Mongo backend limited to LRSes following the LearningLocker schema.

I think we should stay agnostic. The `learninglocker` parser is responsible to extract statements from a leaninglocker Mongo database.

Reply:
As mention in another comment, if we allow custom queries to get documents in database backends, it is no longer required to write dedicated parsers to support various LRS's "schema".

Concerning binary data, @SergioSim will declare an issue to keep in mind considering `python-bsonjs` if we encounter such use-case.